pkg/reconciler/instances/ory/action.go

```diff
@@ -0,0 +1,107 @@
+package ory
+
+import (
+	"context"
+
+	"github.com/kyma-incubator/reconciler/pkg/reconciler"
+	"github.com/kyma-incubator/reconciler/pkg/reconciler/chart"
+	"github.com/kyma-incubator/reconciler/pkg/reconciler/instances/ory/db"
+	"github.com/kyma-incubator/reconciler/pkg/reconciler/instances/ory/jwks"
+	"github.com/kyma-incubator/reconciler/pkg/reconciler/service"
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+	v1 "k8s.io/api/core/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	oryChart     = "ory"
+	oryNamespace = "kyma-system"
+	jwksAlg      = "RSA256"
+	jwksBits     = 2048
+)
+
+type ReconcileAction struct {
+	step string
+}
+
+var (
+	jwksNamespacedName = types.NamespacedName{Name: "ory-oathkeeper-jwks-secret", Namespace: oryNamespace}
+	dbNamespacedName   = types.NamespacedName{Name: "ory-hydra-credentials", Namespace: oryNamespace}
+)
+
+func (a *ReconcileAction) Run(version, profile string, config []reconciler.Configuration, context *service.ActionContext) error {
+	logger := context.Logger
+	component := chart.NewComponentBuilder(version, oryChart).WithNamespace(oryNamespace).WithProfile(profile).WithConfiguration(config).Build()
+	values, err := context.ChartProvider.Configuration(component)
+	if err != nil {
+		return errors.Wrap(err, "failed to retrieve Ory chart values")
+	}
+
+	kubeClient, err := context.KubeClient.Clientset()
+	if err != nil {
+		return errors.Wrap(err, "failed to retrieve native Kubernetes GO client")
+	}
+
+	switch a.step {
+	case "pre-install":
```

Comment:
How about creating an enum for these actions?

Reply:
As they are doing different things, I would suggest to create 2 dedicated Actions for it (separation of concerns).

If these actions are using shared resources, you can use mixins for the shared parts.

Like

     type oryAction struct {
         ...shared things go intos this struct...
      }

     type preAction struct {
         *oryAction
     }
     func (pa *preAction) Run(...) {...}

     type postAction struct {
         *oryAction
     }
     func (a *postAction) Run(...) {...}